custom_components/unifiprotect/binary_sensor.py

```diff
@@ -299,57 +359,4 @@ def _async_fire_motion_event(self):
     @property
     def extra_state_attributes(self):
         """Return the extra state attributes."""
-        key = self.entity_description.key
-        attr = {
-            **super().extra_state_attributes,
-        }
-        if key == _KEY_DARK:
-            return attr
-        if key == _KEY_DOORBELL:
-            attr[ATTR_LAST_TRIP_TIME] = self.device.last_ring.isoformat()
-            return attr
-
-        if isinstance(self.device, Sensor):
-            if key in (_KEY_MOTION, _KEY_DOOR):
-                if key == _KEY_MOTION:
-                    last_trip = self.device.motion_detected_at
-                else:
-                    last_trip = self.device.open_status_changed_at
-
-                attr[ATTR_LAST_TRIP_TIME] = last_trip.isoformat()
-            return attr
-        elif isinstance(self.device, Light):
-            if key == _KEY_MOTION:
-                attr[ATTR_LAST_TRIP_TIME] = self.device.last_motion.isoformat()
-            return attr
-
-        # Camera motion sensors with object detection
-        score = 0
-        if (
-            self.device.is_smart_detected
-            and self.device.last_smart_detect_event is not None
-            and len(self.device.last_smart_detect_event.smart_detect_types) > 0
-        ):
-            score = self.device.last_smart_detect_event.score
-            detected_object = self.device.last_smart_detect_event.smart_detect_types[0]
-            _LOGGER.debug(
-                "OBJECTS: %s on %s",
-                self.device.last_smart_detect_event.smart_detect_types,
-                self._attr_name,
-            )
-        else:
-            if (
-                self.device.is_motion_detected
-                and self.device.last_motion_event is not None
-            ):
-                score = self.device.last_motion_event.score
-            detected_object = None
-
-        attr.update(
-            {
-                ATTR_LAST_TRIP_TIME: self.device.last_motion.isoformat(),
-                ATTR_EVENT_SCORE: score,
-                ATTR_EVENT_OBJECT: detected_object,
-            }
-        )
-        return attr
+        return {**super().extra_state_attributes, **self._attr_extra_state_attributes}
```

Comment:
Entities are either supposed to implement `self._attr_extra_state_attributes` OR `extra_state_attributes` but not both
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L318

Reply:
Technically this is both? Lol. The UFP base entity class overrides calling the HA base class. 

https://github.com/briis/unifiprotect/blob/master/custom_components/unifiprotect/entity.py#L57

Reply:
It only supposed to be handled by the base entity class.

Reply:
I renamed it.